packages/request-node/src/thegraph/TheGraphDataAccess.ts

```diff
@@ -0,0 +1,290 @@
+    const pending = await this.getPending(channelId);
+
+    const result = await this.graphql.getTransactionsByChannelId(channelId);
+
+    return {
+      meta: {
+        transactionsStorageLocation: result.transactions
+          .map((x) => x.hash)
+          .concat(pending.meta.transactionsStorageLocation),
+        storageMeta: this.getStorageMeta(result),
+      },
+      result: {
+        transactions: result.transactions
+          .map(this.getTimestampedTransaction)
+          .concat(pending.result.transactions),
+      },
+    };
+  }
+
+  async getChannelsByTopic(
+    topic: string,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _updatedBetween?: DataAccessTypes.ITimestampBoundaries | undefined,
+  ): Promise<DataAccessTypes.IReturnGetChannelsByTopic> {
+    return this.getChannelsByMultipleTopics([topic]);
+  }
+
+  async getChannelsByMultipleTopics(
+    topics: string[],
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _updatedBetween?: DataAccessTypes.ITimestampBoundaries,
+  ): Promise<DataAccessTypes.IReturnGetChannelsByTopic> {
+    const result = await this.graphql.getChannelsByTopics(topics);
+
+    return {
+      meta: {
+        storageMeta: this.getStorageMeta(result),
+        transactionsStorageLocation: result.transactions.reduce((prev, curr) => {
+          if (!prev[curr.channelId]) {
+            prev[curr.channelId] = [];
+          }
+          prev[curr.channelId].push(curr.hash);
+          return prev;
+        }, {} as Record<string, string[]>),
+      },
+      result: {
+        transactions: result.transactions.reduce((prev, curr) => {
+          if (!prev[curr.channelId]) {
+            prev[curr.channelId] = [];
+          }
+          prev[curr.channelId].push(this.getTimestampedTransaction(curr));
+          return prev;
+        }, {} as DataAccessTypes.ITransactionsByChannelIds),
+      },
+    };
+  }
+
+  async _getStatus(): Promise<any> {
+    return {
+      lastBlock: await this.graphql.getBlockNumber(),
+      storage: await this.storage._getStatus(),
+    };
+  }
+
+  private setPending(
+    channelId: string,
+    transaction: DataAccessTypes.ITransaction,
+    storageResult: StorageTypes.IAppendResult,
+  ) {
+    this.pending[channelId] = { transaction, storageResult };
+  }
+
+  private async getPending(channelId: string): Promise<DataAccessTypes.IReturnGetTransactions> {
+    const emptyResult = {
+      meta: {
+        transactionsStorageLocation: [],
+        storageMeta: {},
+      },
+      result: {
+        transactions: [],
+      },
+    };
+    if (!this.pending[channelId]) {
+      return emptyResult;
+    }
+    const { storageResult, transaction } = this.pending[channelId];
+
+    const { transactions } = await this.graphql.getTransactionsByHash(
+      this.pending[channelId].storageResult.id,
+    );
+
+    // if the pending tx is found, remove its state and fetch the real data
+    if (transactions.length > 0) {
```

Comment:
What if for the same request/channelId, there are 2 transactions, one confirmed and one pending?

Reply:
nice catch ;) this actually happens a lot! It's handled in `getTransactionsByChannelId` where we merge pending and not pending

Reply:
Did you skip this edge case in the end? Maybe I missed something

Reply:
no, it's already handled in another method, no need to change (I think! feel free to double check)